### PR TITLE
Embeddings: HF model hub support added via transformers 

### PR DESCRIPTION
**Pull request by raghavdixit99**
Added hf_model hub functions to generate text embeddings , unit test , minor change to `pyproject.toml` to fix potential ruff isorting issues 

Todo in next update-

-  add mistral client support 
- add functionality to support all kinds of embeddings models [ multimodal ] , right now we have different models as well but soon would like to add all types of multimodal support via transformers within one script so user can just provide name and load it

### CODE REVIEW

**Comment by raghavdixit99** [comment]
Thank you @AyushExel, have resolved the review changes, please check and merge.

**Comment by AyushExel** [comment]
Looks good. Write docs and a separate callout for colbert

File: python/python/lancedb/embeddings/transformers.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+import numpy as np
+
+from ..util import attempt_import_or_raise
+from .base import EmbeddingFunction
+from .registry import register
+from .utils import TEXT
+
+
+@register("huggingface")
```
Comment (AyushExel) [inline python/python/lancedb/embeddings/transformers.py:25]: add this to __init__ file so that this actually gets registered

**Comment by raghavdixit99** [comment]
@AyushExel review pls

**Comment by AyushExel** [comment]
Please fix test. Pydantic 1.x compat related error

**Review by AyushExel — APPROVED** [review]

**Pull request merged**